src/Eccube/Entity/OrderItem.php

```diff
@@ -30,52 +31,34 @@ class OrderItem extends \Eccube\Entity\AbstractEntity implements ItemInterface
 {
     use PointRateTrait;
 
-    private $price_inc_tax = null;
-
-    /**
-     * Set price IncTax
-     *
-     * @param  string       $price_inc_tax
-     *
-     * @return OrderItem
-     */
-    public function setPriceIncTax($price_inc_tax)
-    {
-        $this->price_inc_tax = $price_inc_tax;
-
-        return $this;
-    }
-
     /**
      * Get price IncTax
      *
-     * @return string
+     * @return integer
      */
     public function getPriceIncTax()
     {
-        return $this->price_inc_tax;
+        $TaxType = $this->getTaxType();
+        $TaxDisplayType = $this->getTaxDisplayType();
+        if (is_object($TaxType) && is_object($TaxDisplayType)) {
+            // 課税かつ外税の場合にのみ商品単価に消費税を足した金額を返す
+            if ($TaxType->getId() == TaxType::TAXATION && $TaxDisplayType->getId() == TaxDisplayType::EXCLUDED) {
```

Comment:
@okazy -san,
Because class `PriceChangeValidator` validate base on  `isProduct` function.

https://github.com/EC-CUBE/ec-cube/blob/experimental/sf/src/Eccube/Service/PurchaseFlow/Processor/PriceChangeValidator.php#L34

However, this line check base on the Tax Type and Tax Display Type, it will not match.

Q: How can I resolve it?
P.S: why don't you use `isProduct` function?